README.md

```diff
@@ -119,7 +119,7 @@ Any and all settings which affect the behavior of the generative plugin should b
 - `githubWorkflowBuildPreamble` : `Seq[WorkflowStep]` – Contains a list of steps which will be inserted into the `build` job in the **ci.yml** workflow *after* setup but *before* the `sbt test` invocation. Defaults to empty.
 - `githubWorkflowBuildPostamble` : `Seq[WorkflowStep]` – Similar to the `Preamble` variant, this contains a list of steps which will be added to the `build` job *after* the `sbt test` invocation but before cleanup. Defaults to empty.
 - `githubWorkflowBuild` : `Seq[WorkflowStep]` – The steps which invoke sbt (or whatever else you want) to build and test your project. This defaults to just `[sbt test]`, but can be overridden to anything. For example, sbt plugin projects probably want to redefine this to be `Seq(WorkflowStep.Sbt(List("test", "scripted")))`, which would run the `test` and `scripted` sbt tasks, in order. Note that all uses of `WorkflowStep.Sbt` are compiled using the configured `githubWorkflowSbtCommand` invocation, and properly configured with respect to the build matrix-selected Scala version.
-- `githubWorkflowJavaVersions` : `Seq[JavaSpec]` – A list of Java versions to be used for the build job. The publish job will use the *first* of these versions. Defaults to `JavaSpec.temurin("11")`).
+- `githubWorkflowJavaVersions` : `Seq[JavaSpec]` – A list of Java versions to be used for the build job. The publish job will use the *first* of these versions. Defaults to `JavaSpec.temurin("17")`).
```

Comment:
Thanks for clarifying. I still think this PR is useful as we should be using the latest default LTS version by default which is 17, it also updates the GraalVM to latest version.

Reply:
> I still think this PR is useful as we should be using the latest default LTS version by default which is 17

Note that not everyone feels this way.
- https://github.com/sbt/sbt-github-actions/issues/96

Reply:
My counter argument to that would be that if you don't want the plugin to update to the latest stable in a newer release than you should set `githubWorkflowJavaVersions` statically (just like this plugin itself does when publishing as we found out) and you will never get this problem.

Updating to the latest stable LTS release is completely reasonable in my view, people also need to move away from JDK versions especially ones that aren't even supported anymore.

Reply:
I do think however this behaviour should be documented so I will update the README.md to signify this.

Reply:
Assuming the target audience of this plugin is library authors, they need to be aware of what the downstream impact is when they adopt or upgrade the plugin. If we make the change, I do agree that we should yell from the mountaintops as "changes with compatibility implication".

Reply:
If by "yell by the mountaintops" you mean mentioning it in release notes as well as in README.md that the default JDK/Scala version can change, then definitely yes (this to me is already implied).

Reply:
Yea, I guess generally highlighting in the readme (here's a PR - https://github.com/sbt/sbt-github-actions/pull/142), and in the release note as the top item I guess.

Reply:
There's a difference in what's most appropriate for applications (17) and libraries (we're now in the gray zone between 8 and 11).  That's why typelevel-nix maintains two devshells.

I would very strongly recommend against this change.  If you guess too low, it usually breaks at compile time.  If you guess too high, it usually breaks at runtime, often in another org.

Reply:
> There's a difference in what's most appropriate for applications (17) and libraries (we're now in the gray zone between 8 and 11). That's why typelevel-nix maintains two devshells.

I was talking about libraries, not applications. For libraries that are very complex then yes, setting `githubWorkflowJavaVersions` makes sense (and this has always been possible). For most "standard" libraries, I would actually argue its beneficial for JDK to update to latest stable release (once its been known to work with Scala, it wouldn't be updated in this plugin beforehand).

> I would very strongly recommend against this change. If you guess too low, it usually breaks at compile time.

What is the issue in setting `githubWorkflowJavaVersions` if you have such strict requirements? If it was impossible to change this behaviour I would understand, but setting `githubWorkflowJavaVersions` is also a lot more clear about what your intention is, i.e. its ultra critical that my library does not update the JVM/Scala version when sbt-github-actions.

> If you guess too high, it usually breaks at runtime, often in another org.

Unless I am missing something, we are talking about CI JVM version, not the JVM version that a user is running that happens to have a transitive dependency.

Reply:
I guess the supposed scenario here is that an open source project uses this plugin without setting `githubWorkflowJavaVersions` and with automated publishing merges Scala Steward suggesting to bump up to then ext version (0.15.0?), and it would then publish the library at JDK 17. Since the production Java runtime is set at different companies etc, the downstream implication is that in those companies the library would no longer be usable. One solution for that is (sbt-multi-release-jar - https://github.com/eed3si9n/scalacenter-proposal/discussions/1), but for now libraries tend to pick either JDK 8 or 11.

Generally, if we automate something, I think it's better for the automation to be least surprising.

Reply:
> I guess the supposed scenario here is that an open source project uses this plugin without setting githubWorkflowJavaVersions and with automated publishing merges Scala Steward suggesting to bump up to then ext version (0.15.0?), and it would then publish the library at JDK 17. Since the production Java runtime is set at different companies at their own pace etc, the downstream implication is that in those companies the library would no longer be usable. One solution for that is (sbt-multi-release-jar - https://github.com/eed3si9n/scalacenter-proposal/discussions/1), but for now libraries tend to pick either JDK 8 or 11.

In that case would changing the default value of `githubWorkflowJavaVersions` to `Seq(JavaSpec.temurin("8"), JavaSpec.temurin("17"))` with only the second value being updated as sbt-github-actions is updated and since `JavaSpec.temurin("8")` is first that is the version that will be published make more sense?

Reply:
> JavaSpec.temurin("8") is first that is the version that will be published make more sense?

That sounds great.